src/execution/operator/aggregate/physical_window.cpp

```diff
@@ -789,21 +812,48 @@ idx_t PhysicalWindow::GetBatchIndex(ExecutionContext &context, DataChunk &chunk,
 	return lstate.batch_index;
 }
 
+bool WindowGlobalSourceState::UpdateBlockedTasks(bool blocked, InterruptState &interrupt_state) const {
+	lock_guard<mutex> guard(lock);
+	if (blocked) {
+		blocked_tasks.push_back(interrupt_state);
+	} else {
+		// The pipeline is unblocked, so flush tasks
+		for (auto &state : blocked_tasks) {
+			state.Callback();
+		}
+		blocked_tasks.clear();
+	}
+	return blocked;
+}
+
 SourceResultType PhysicalWindow::GetData(ExecutionContext &context, DataChunk &chunk,
                                          OperatorSourceInput &input) const {
 	auto &gsource = input.global_state.Cast<WindowGlobalSourceState>();
 	auto &lsource = input.local_state.Cast<WindowLocalSourceState>();
-	while (chunk.size() == 0) {
-		//	Move to the next bin if we are done.
-		while (!lsource.scanner) {
-			if (!lsource.NextPartition()) {
-				return chunk.size() > 0 ? SourceResultType::HAVE_MORE_OUTPUT : SourceResultType::FINISHED;
-			}
-		}
 
-		lsource.Scan(chunk);
-		gsource.returned += chunk.size();
+#if 1
```

Comment:
This part is still a TODO I think?

Reply:
Ah I just read that part of the PR description, apologies

Reply:
We had some difficulty getting this to work properly in the aggregate operators
One issue I see is that unblocking could be happening in too few places, resulting in blocking the last task that runs, with no tasks left to unblock them

Reply:
This might help to look at:
https://github.com/duckdb/duckdb/pull/11528